source/common/upstream/eds.cc

```diff
@@ -62,26 +62,31 @@ void EdsClusterImpl::BatchUpdateHelper::batchUpdate(PrioritySet::HostUpdateCb& h
   const uint32_t overprovisioning_factor = PROTOBUF_GET_WRAPPED_OR_DEFAULT(
       cluster_load_assignment_.policy(), overprovisioning_factor, kDefaultOverProvisioningFactor);
 
+  const HostVector empty_hosts;
+  LocalityWeightsMap empty_locality_map;
+
   // Loop over all priorities that exist in the new configuration.
   auto& priority_state = priority_state_manager.priorityState();
   for (size_t i = 0; i < priority_state.size(); ++i) {
+    if (parent_.locality_weights_map_.size() <= i) {
+      parent_.locality_weights_map_.resize(i + 1);
+    }
     if (priority_state[i].first != nullptr) {
-      if (parent_.locality_weights_map_.size() <= i) {
-        parent_.locality_weights_map_.resize(i + 1);
-      }
       cluster_rebuilt |= parent_.updateHostsPerLocality(
           i, overprovisioning_factor, *priority_state[i].first, parent_.locality_weights_map_[i],
           priority_state[i].second, priority_state_manager, updated_hosts);
+    } else {
+      // if the new update contains a priority with no hosts, call update with an empty set of hosts
+      cluster_rebuilt |= parent_.updateHostsPerLocality(
+          i, overprovisioning_factor, empty_hosts, parent_.locality_weights_map_[i],
```

Comment:
can you just create the `empty_hosts`/`empty_locality_map`inline? i.e. just do `{}` or `HostVector{}` instead of using a variable

Reply:
I get an error when I try to do this for empty_locality_map, but it looks like empty_hosts can be converted.